### ISSUE TITLE
[Discussion] Mapgen

### ISSUE THREAD

**Issue by wrynvera** [Craigs-Crew/edgy-dark-ominous-game#75]
Starting this discussion on mapgen because it is quite a big concern that we haven't had a serious discussion on yet.

Presently, we are presented with these options for terrain, as previous discussions have suggested, if you do have any suggestions for better alternatives however, do state below, and ill consolidate them to this main list here.
With regards to biome, the consensus is to rewrite them, but there's no concrete idea on what to include or add yet. 

Feel free to discuss/suggest in discord as well, but do ping me, and ill transcribe to this list accordingly.
This is not an approval list, nor WIP/concept list, just a consolidation of possible directions.

Terrain
1. Keep v6
2. Carpathian 
3. Carpathian with noise parameters modifications (Largely leaning towards)
4. Other base mapgen with noise parameters modifications
5. Lua mapgen 
6. Builtin mapgen (C++)

Biomes
- ?

**Comment by mt-frontier** [comment]
My two cents
Terrain: 
V6 doesn't give flexibility with biomes or enough variety in terrain to keep my interest anymore as a player although it is builder friendly and has nostalgic appeal, I admit.
Carpathian, I think, is the most builder friendly out of the box modern mapgen that accommodates a variety of tastes in landscape (flat vs mountains) and its naturalistic terrain is appealing. 
V7 personally I don't care for this mapgen by out of the box except for the river cliffs but that is not really a compelling enough reason to use it for a building server. 
 
While it may be possible, I have my doubts that shooting in the dark trying to find the right combination of noise params settings with V7 is worth the extra time map testing that I think would be required to ensure a consistent builder friendly map that everyone is happy with, let alone having a predictable result across map seeds. v7 is the most flexible but this fact makes it too wild and unpredictable imo. Since Carpathian achieves >80% of what i'm looking for in a map for building out-of-the-box and v7 is nowhere near that it makes much more sense to me to start there, test map seeds, and then tweak noise params and biomes as needed. Possibly cliffs can be acheived in Carpathian?

Lua mapgen: this could be interesting if we wanted to take an single-node approach where a contiguous map is not essential such as having areas of map to support variations in mapgens in different parts of the map/world. Ultimately, though, I don't think this will appeal to the majority of players that just want to wander and find cool places to build and will be a lot of extra work to negotiate and develop and strays quite a bit from some of the appealing aspects of a relaxed open world building game such as  Craig's.

Tl; dr: I vote Carpathian (with tweaks as needed/agreed upon) I would be open to lua mapgen or discussion around using the underground or "space" areas of the world in a more interesting way, but I think that borders on being a separate issue than the base mapgen if one of the stated goals is to keep things simple and easy to replicate.

Biomes:
I'm good with cutting and tweaking biomes to favor grasslands and deciduous forest to a certain extent to simulate v6, but I like some biome diversity and removing biomes carelessly can lead to odd border regions such as having snowy biomes next to jungle. Adjusting shore biomes elevations to reduce shoreline sand is okay with me. I'd be good with removing permafrost biome as this is at the extreme of the biomes and should not impact the map too much. I also agree with either removing/reducing dune biomes and removing dune grass decoration at a minimum. Personally, I'd prefer not to get into adding new biomes unless they exist underground or they are floating or in "outer space."

**Comment by theraven262** [comment]
I agree we should go for carpathian, for these same reasons.
I don't want to remove sandstone deserts because I think that there is a lot of potential in enriching them with ores and new decorations (such as amber pillars or something).
I'm currently working on oregen, and am making stuff generate in more than just stone. Sandstone and its derivatives will be much more fun to mine.

**Comment by ChandlerBing1994** [comment]
Agreed with Lumber, we should really stick with Carpathian. If we can get it to be 70% Carpathian and 30% v6 to have that nice cozy Craig's feel I'd love it. Removal of dune biomes and permafrost, reducing the shoreline and removing marram grass, etc. etc. would be great. I'd also suggest making mountain generation more grounded in reality, meaning no floating chunks.

**Comment by lumberjackgames** [comment]
I think the Carpathian terraced mountains have some of the good qualities of v6. Perhaps looking into increasing their prevalence over the more free form type mountains. This could be an option that would make the map overall more builder friendly and v6 like. This may require some engine modifications in the Carpathian code but I'd consider it at least worth exploring to see if it would be feasible if other members are interested.

**Comment by theraven262** [comment]
> I think the Carpathian terraced mountains have some of the good qualities of v6. Perhaps looking into increasing their prevalence over the more free form type mountains. This could be an option that would make the map overall more builder friendly and v6 like. This may require some engine modifications in the Carpathian code but I'd consider it at least worth exploring to see if it would be feasible if other members are interested.

Tweaking [this](https://github.com/minetest/minetest/blob/master/minetest.conf.example#L2629) noise should do it I think.

**Comment by wrynvera** [comment]
Engine modifications is a last resort, I rather not maintain our own fork of minetest.
If we want to consider large differences that modifications of existing mapgen noises can't do the trick, I would rather head in the direction of a custom lua mapgen - admittedly is slower, but the prior is too much effort imo.

**Comment by wrynvera** [comment]
Closing this as it seems like the discussion has mostly moved to the discord channel and the majority of the things as stated here are already WIP or completed.
